### ISSUE TITLE
Migration of logging package has parameter to LogLevel.== made nullable, incorrectly

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/sdk#39625]
--- snipped from an offline comment by @bwilkerson ---

... we noticed that the parameter to `LogLevel.==` had been made nullable when it shouldn't have been. I can't reproduce that behavior on master, but wondered yesterday if the engine was getting the wrong SDK or was not correctly handling code from the SDK.

--- snip ---

**Comment by bwilkerson** [comment]
@srawlins

**Comment by srawlins** [comment]
Confirmed this happens at HEAD this morning when using the non-NNBD SDK. Not sure what is happening in the study...

To be precise, I build two SDKs:

1. `./tools/build.py --mode release --arch x64 runtime create_sdk`

   This creates an SDK at `xcodebuild/ReleaseX64` which is functional

   Then I set `STUDY_SDK=$PWD/xcodebuild/ReleaseX64/dart-sdk`
2. `./tools/build.py --mode release --arch x64 runtime create_sdk --nnbd`

   This creates an SDK at `xcodebuild/ReleaseX64NNBD` which contains the NNBD core libs.

   Then I set `STUDY_NNBD_SDK=$PWD/xcodebuild/ReleaseX64NNBD/dart-sdk`

I migrate the logging package with this command:

```shell
$STUDY_SDK/bin/dart ~/code/dart-sdk/sdk/pkg/dartfix/bin/dartfix.dart \
    --fix=non-nullable --preview-dir nnbd-i13n --sdk $STUDY_NNBD_SDK .
```

When I run this, the type of the parameter of `LogLevel.==` does not change to be nullable. If I omit the `--sdk` argument, then the parameter of `LogLevel.==` does change to be nullable.

I don't think there is anything else to do here.
